### PR TITLE
Extend example docs with OpenOCD commands and add FAQ

### PR DESCRIPTION
**Pull request by forGGe**
According to the feedback from Manuel, the documentation is extended with FAQ and examples are clarified.

### CODE REVIEW

File: doc/sphinx/source/guides/running-openocd-without-sudo.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
+.. _theCore_OpenOCD_NoRoot:
+
+Running OpenOCD without root privileges
```
Comment (rasendubi) [inline doc/sphinx/source/guides/running-openocd-without-sudo.rst]: The whole guide, NixOS version :smirk: 
```nix
{
  users.extraGroups.plugdev = { };
  users.extraUsers.rasen.extraGroups = [ "plugdev" "dialout" ];

  services.udev.packages = [ pkgs.openocd ];
}
```

Comment (forGGe) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: Could you please gimme a link or extend this message, so I can post it in this page, too? :)

Comment (rasendubi) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: Basically...

That's how you create a group:
```nix
{ users.extraGroups.plugdev = { }; }
```

That's how you add a user to the group:
```nix
{ users.extraUsers.rasen.extraGroups = [ "plugdev" ]; }
```

And that's how you install udev rules that come with `openocd` package:
```nix
{ services.udev.packages = [ pkgs.openocd ]; }
```

Comment (forGGe) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: So, where to add those lines?

Comment (rasendubi) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: NixOs configuration file (default, `/etc/nixos/configuration.nix`)

Comment (rasendubi) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: There is only one configuration file for NixOS ;)

Comment (forGGe) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: Got it!

File: doc/sphinx/source/guides/getting-started.rst [hunk 2]
```diff
@@ -67,7 +67,7 @@ Using Nix
 #. Run Nix shell to get all environment required::
 
      # Run from theCore directory
-     nix-shell --pure
+     nix-shell .
```
Comment (rasendubi) [inline doc/sphinx/source/guides/getting-started.rst]: `.` is not required

Comment (forGGe) [reply doc/sphinx/source/guides/getting-started.rst]: Let's leave it there to be more explicit about importance of current working dir.

File: doc/sphinx/source/guides/running-openocd-without-sudo.rst [hunk 3]
```diff
@@ -4,7 +4,24 @@ Running OpenOCD without root privileges
...
+NixOS Linux
+~~~~~~~~~~~
+
+Add following lines to your ``/etc/nixos/configuration.nix``:
+
+.. code-block:: nix
+
+  {
+    users.extraGroups.plugdev = { };
+    users.extraUsers.rasen.extraGroups = [ "plugdev" "dialout" ];
```
[the remark this answers is not in the document]
Comment (forGGe) [reply doc/sphinx/source/guides/running-openocd-without-sudo.rst]: Done, lol

**Pull request merged**